src/utilities/scalarValueFromAST.js

```diff
@@ -0,0 +1,43 @@
+/* @flow */
+/**
+ *  Copyright (c) 2017, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+import keyValMap from '../jsutils/keyValMap';
+import * as Kind from '../language/kinds';
+import type { ValueNode } from '../language/ast';
+
+/**
+ * Create a JavaScript value from a GraphQL languate AST representation
+ * of Scalar.
+ */
+export function scalarValueFromAST(astValue: ValueNode): mixed {
```

Comment:
If this function is for scalars, then it should likely not support LIST or OBJECT, correct?